### PR TITLE
Fix HTML entity escaping in Glimmer printer

### PR DESCRIPTION
**Pull request by omghax**
Previously, prettier would unescape `&lt;` and `&gt;` which would result in invalid HTML. This change escapes Glimmer TextNode nodes to ensure that `&`, `<`, and `>` remain escaped properly.

### CODE REVIEW

**Comment by lydell** [comment]
```
&amp;
&nbsp;
&#160;
&#xa0;
```

File: src/language-handlebars/printer-glimmer.js [hunk 1]
```diff
@@ -28,6 +28,21 @@ const voidTags = [
...
+// Escaping logic based on handlebars.js:
+// https://github.com/wycats/handlebars.js/blob/master/lib/handlebars/utils.js
+const escape = {
+  "&": "&amp;",
+  "<": "&lt;",
+  ">": "&gt;",
+  "\xa0": "&nbsp;"
+};
+
+const badChars = /[&<>\xa0]/g;
```
Comment (lydell) [inline src/language-handlebars/printer-glimmer.js]: So you had to add yet another character here? Where will it end?

File: tests/html_glimmer/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -426,6 +426,14 @@ exports[`else-if.hbs 1`] = `
 {{/if}}
 `;
 
+exports[`escaped-chars.hbs 1`] = `
+<p>Some escaped characters: &lt; &gt; &amp; &nbsp; &#160; &#xa0;</p>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<p>
+  Some escaped characters: &lt; &gt; &amp; &nbsp; &nbsp; &nbsp;
```
Comment (lydell) [inline tests/html_glimmer/__snapshots__/jsfmt.spec.js.snap]: So all of those are turned into `&nbsp;`? Not sure if that’s a feature or a bug. Perhaps it’s out of scope for this PR.

**Comment by lydell** [comment]
This doesn’t feel like the right thing to do overall.

Perhaps we could merge a quick-fix for only `<`, `>` and `&` so that at least the output code is valid, and then open an issue about properly handling escapes.

**Comment by omghax** [comment]
@lydell Okay, I'll remove that commit from this PR and just leave it focused on `<`, `>`, and `&`.

**Comment by rwjblue** [comment]
FWIW - `@glimmer/syntax` uses `simple-html-tokenizer` which ships with a file with all the HTML char references ([see here](https://github.com/tildeio/simple-html-tokenizer/blob/master/src/generated/html5-named-char-refs.ts)) which is actually _generated_ from https://github.com/w3c/html/blob/master/entities.json...

**Comment by j-f1** [comment]
What’s the status of this? Should we merge or close?

**Comment by lifeart** [comment]
@omghax @j-f1 sorry for pinging, is it possible to get it merged?

**Comment by j-f1** [comment]
@lydell what do you think?

**Comment by lydell** [comment]
I guess we could merge it.

**Comment by j-f1** [comment]
Can you fix the merge conflicts @omghax?

**Comment by billdami** [comment]
Just curious on the status of this issue? Also, it was a bit hard to judge from the discussion here, but will this fix result in `&nbsp;` _not_ being converted to literal spaces? We are seeing issues with these specifically, since `&nbsp;` behaves differently in the DOM then a normal space (e.g. collapsing multiple adjacent spaces).

Ideally, we'd love a way to disable prettier's entity conversion entirely (for example we prefer to see/edit `&raquo;` in the source rather than `»`), but non-breaking spaces was the more immediate concern.

**Comment by evilebottnawi** [comment]
@billdami looks like PR was abandoned, feel free to send new PR

**Comment by rwjblue** [comment]
FWIW, new changes were made to the latest @glimmer/syntax package to allow a "codemod" mode. In that mode, **no** entity escaping is done at all. If we migrate to that new mode, I think we will have a much better time.

See https://github.com/glimmerjs/glimmer-vm/pull/938 for the changes I'm referring to...

**Comment by lydell** [comment]
@rwjblue Awesome! We should definitely try that out.

**Comment by GavinJoyce** [comment]
I've created https://github.com/prettier/prettier/pull/6234 which uses the new "codemod" glimmer mode

**Comment by GavinJoyce** [comment]
@evilebottnawi perhaps you meant to approve https://github.com/prettier/prettier/pull/6234 instead? I think this PR can then be closed

**Comment by lydell** [comment]
Supersedes by #6234.

**Pull request closed without merging**